helpers/import.js

```diff
@@ -38,87 +46,127 @@ function importTable(gtfs, tableName) {
+  return {
+    keys,
+    rowsSlices,
+  };
 }
 
-function processGtfsTableRows(gtfs, tableName, keys, rows, indexKeys, GtfsRow) {
+function processGtfsTable(gtfs, keys, rowsSlices, tableName, indexKeys) {
   let table = (indexKeys.setOfItems) ? new Set() : new Map();
 
-  const regexPatternObjects = gtfs._regexPatternObjectsByTableName.get(tableName);
-
-  rows.forEach((row) => {
-    if (regexPatternObjects) {
-      row = applyRegexPatternObjectsByTableName(regexPatternObjects, keys, row, tableName);
-    }
+  if (rowsSlices === undefined || rowsSlices === null || rowsSlices.length === 0) {
+    return table;
+  }
 
-    const trimmedRow = row.map(value => value.trim());
-    const gtfsRow = new GtfsRow(trimmedRow);
+  const parsedKeys = Papa.parse(keys, { delimiter: ',', skipEmptyLines: true });
+  const trimmedKeys = parsedKeys.data[0].map(key => key.trim());
+  checkThatKeysIncludeIndexKeys(trimmedKeys, indexKeys, tableName);
 
-    if (indexKeys.indexKey) {
-      table.set(gtfsRow[indexKeys.indexKey], gtfsRow);
-    } else if (indexKeys.firstIndexKey && indexKeys.secondIndexKey) {
-      if (table.has(gtfsRow[indexKeys.firstIndexKey]) === false) {
-        table.set(gtfsRow[indexKeys.firstIndexKey], new Map());
-      }
+  const GtfsRow = createGtfsClassForKeys(trimmedKeys);
+  let errorMessage;
 
-      table.get(gtfsRow[indexKeys.firstIndexKey]).set(gtfsRow[indexKeys.secondIndexKey], gtfsRow);
-    } else if (indexKeys.singleton) {
-      table = gtfsRow;
-    } else if (indexKeys.setOfItems) {
-      table.add(gtfsRow);
+  eachWithLog(`Importation:${tableName}`, rowsSlices, (rowsSlice) => {
+    if (!rowsSlice || !rowsSlice.trim) {
+      return;
     }
-  });
 
-  return table;
-}
+    rowsSlice = rowsSlice.trim();
 
-function applyRegexPatternObjectsByTableName(regexPatternObjects, keys, row, tableName) {
-  const rowStringified = String(row);
-  let modifiedRowStringified = rowStringified;
+    const parsedRow = Papa.parse(`${keys}${rowsSlice}`, { delimiter: ',', skipEmptyLines: true });
 
-  regexPatternObjects.forEach(({ regex, pattern }) => {
-    modifiedRowStringified = rowStringified.replace(regex, pattern || '');
+    if (parsedRow.errors.length) {
+      if (!errorMessage) {
+        errorMessage = `Invalid rows in table ${tableName}:\n`;
+      }
 
-    if (modifiedRowStringified !== rowStringified) {
-      process.notices.addInfo(
-        'Applying Changes on Raw GTFS',
-        `Applying regex replace to table: "${tableName}". regex: "${regex}".`
-      );
+      parsedRow.errors.forEach((error) => {
+        errorMessage += `Line: ${error.row}
+Issue: ${error.message}
+Row: ${parsedRow.data[error.row].join(',')}`;
+      });
     }
-  });
 
-  const parsedModifiedRow = Papa.parse(`${keys}\n${modifiedRowStringified}`, {
-    delimiter: ',',
+    const [, ...rows] = parsedRow.data; // we don't need the header row, it's already stored in parsedKeys/trimmedKeys
```

Comment:
I created this test: https://jsperf.com/destructuring-vs-array-shift-to-get-rows

we can:
```js
parsedRow.data.shift()
   .forEach((row) => {
```
or 
we can also loop directly
```js
    for (let i = 1; i > arrray.length; i++) {
      const row = rows[i];
    }
```

Reply:
I updated the code using the for loop starting at i = 1